File: CSharpCodeBase/aiactions/scanenemies.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

local AI_ScanEnemies = Class({__includes=AINode, init=function(self, entity)
  AINode.init(self, entity)
end})

function AI_ScanEnemies:Visit()
  if self.status == NODE_READY then
    self.entity:ScanEnemies()
    self.status = NODE_FAILURE -- keep it failure to running tree
  end
  self:Sleep(0.3)
  return self.status
end

return AI_ScanEnemies

File: CSharpCodeBase/aiactions/scanenemies.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class scanenemies :ainode{
 var AI_ScanEnemies = Class({__includes=AINode, init=function(self, entity)
   AINode.init(self, entity);
 }})
 public void AI_Visit(){
   if(this.status == NODE_READY  ){ 
     this.entity:ScanEnemies();
     this.status = NODE_FAILURE // keep it failure to running tree
   }
   self:Sleep(0.3);
   return this.status;
 }
 return AI_ScanEnemies}}